Storage/src/Connection/Rest.php

```diff
@@ -628,6 +629,157 @@ private function sendWithRetry($resource, $method, $args)
+            $valueToAdd
+        );
+
+        $valueToAdd = sprintf("gccl-attempt-count/%s", $currentAttempt);
+        $this->updateHeader(
+            'x-goog-api-client',
+            $arguments,
+            $valueToAdd,
+            false
+        );
+    }
+
+    /**
+     * Amends the given header key with new value for a request such that
+     * the $request headers aren't modified directly and instead $options array
+     * which are applied to the request just before sending it at core level.
+     * Thus the $request object remains the same between each retry request at
+     * RequestWrappers' level.
+     *
+     * @param string $headerLine The header line to update.
+     * @param array &$arguments The arguments array(passed by reference) used by
+     * execute method of ExponentialBackoff object.
+     * @param string $value The value to be ammended in the header line.
+     * @param bool $getHeaderFromRequest [optional] A flag which determines if
+     *  existing header value is read from $request or from $options. It's useful
+     *  to read from $options incase we update multiple values to a single
+     *  header key.
+     */
+    private function updateHeader(
+        string $headerLine,
+        array &$arguments,
+        string $value,
+        bool $getHeaderFromRequest = true
+    ): void {
+        // Fetch request and options
+        $request = $this->fetchRequest($arguments);
+        $options = $this->fetchOptions($arguments);
+
+        // Create the modified header
+        $headerValue = '';
+        if ($getHeaderFromRequest) {
+            $headerValues = $request->getHeader($headerLine);
+            $headerValues[] = $value;
+            $headerValue = implode(' ', $headerValues);
+        } else {
+            $headerValue = $options['headers'][$headerLine];
+            $headerValue .= (' ' . $value);
+        }
+
+        // Amend the $option's header value
+        $options['headers'][$headerLine] = $headerValue;
+
+        // Set the $argument's options array
+        $this->setOptions($arguments, $options);
+    }
+
+
+    /**
+     * This helper method fetches Request object from the $argument list.
+     * @param mixed $arguments
+     * @return Request|null
+     */
+    private function fetchRequest($arguments): Request|null
+    {
+        $request = null;
+        foreach ($arguments as $argument) {
+            if ($argument instanceof Request) {
+                $request = $argument;
+            }
+        }
+        return $request;
+    }
+
+    /**
+     * This helper method fetches $options array from the $argument list.
+     * @param mixed $arguments
+     * @return array
+     */
+    private function fetchOptions($arguments): array
+    {
+        $options = [];
+        foreach ($arguments as $argument) {
+            if (is_array($argument) && isset($argument['headers'])) {
+                $options = $argument;
+            }
+        }
+        return $options;
```

Comment:
```suggestion
        foreach ($arguments as $argument) {
            if (is_array($argument) && isset($argument['headers'])) {
                return $argument;
            }
        }
        return [];
```

Reply:
Accepted the changes in [2014c4c](https://github.com/googleapis/google-cloud-php/pull/5741/commits/2014c4c77cbde0ccca4e7b3ff7b041301f5cf01b)

Thanks